login.go

```diff
@@ -0,0 +1,202 @@
+package conductoroneapi
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type ClientCredentials struct {
+	ClientID     string
+	ClientSecret string
+}
+
+type DeviceCodeResponse struct {
+	DeviceCode      string `json:"device_code"`
+	UserCode        string `json:"user_code"`
+	VerificationURI string `json:"verification_uri_complete"`
+	ExpiresIn       int64  `json:"expires_in"`
+	Interval        int64  `json:"interval"`
+}
+
+func LoginFlow(ctx context.Context, tenantName string, clientID string) (*ClientCredentials, error) {
+	opts := []SDKOption{}
+	// If they pass a URL, use the whole URL
+	if strings.Contains(tenantName, ".") {
+		opts = append(opts, WithServerURL(tenantName))
+	} else {
+		opts = append(opts, WithTenantDomain(tenantName))
+	}
+	client := New(opts...)
+
+	codeResp, err := getDeviceCode(ctx, client, clientID)
+	if err != nil {
+		return nil, fmt.Errorf("error getting device code: %w", err)
+	}
+
+	tokenResp, err := doTokenRequest(ctx, client, clientID, codeResp)
+	if err != nil {
+		return nil, fmt.Errorf("error doing token request: %w", err)
+	}
+
+	clientCredential, err := doClientCredentialRequest(ctx, client, tokenResp)
+	if err != nil {
+		return nil, fmt.Errorf("error doing client credential request: %w", err)
+	}
+
+	return &ClientCredentials{
+		ClientID:     clientCredential.Client.ClientID,
+		ClientSecret: clientCredential.ClientSecret,
+	}, nil
+}
+
+func getDeviceCode(ctx context.Context, client *ConductoroneAPI, clientID string) (*DeviceCodeResponse, error) {
+	httpClient := client.sdkConfiguration.DefaultClient
+
+	deviceCodeURL := "https://" + client.sdkConfiguration.ServerURL + "/auth/v1/device_authorization"
+	vals := url.Values{}
+	vals.Add("client_id", clientID)
+
+	req, err := http.NewRequestWithContext(ctx, "POST", deviceCodeURL, strings.NewReader(vals.Encode()))
```

Comment:
do we have a common constructor for requests that should apply headers like user-agent?